umh-core/pkg/communicator/api/v2/http/common.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2025 UMH Systems GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package http
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-retryablehttp"
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/communicator/pkg/tools/safejson"
+	"go.uber.org/zap"
+)
+
+// Endpoint represents the HTTP path type used for API endpoints.
+type Endpoint string
+
+var (
+	// LoginEndpoint is the login endpoint for instance authentication.
+	LoginEndpoint Endpoint = "/v2/instance/login"
+	// PushEndpoint is the push endpoint for sending instance data.
+	PushEndpoint Endpoint = "/v2/instance/push"
+	// PullEndpoint is the pull endpoint for retrieving instance data.
+	PullEndpoint Endpoint = "/v2/instance/pull"
+)
+
+// DoHTTPRequestWithRetry performs HTTP requests with automatic retries for connection errors.
+// This is the main function that external callers should use for HTTP requests.
+func DoHTTPRequestWithRetry[T any](ctx context.Context, method, url string, data *T, header map[string]string, cookies *map[string]string, insecureTLS bool, enableLongPoll bool, logger *zap.SugaredLogger) (*http.Response, error) {
+	retryClient := retryablehttp.NewClient()
+	retryClient.HTTPClient = GetClient(insecureTLS)
+	retryClient.RetryMax = 10
```

Comment:
Maybe move these constants also into a constant.go package, maybe inside even the same package here, but then you can also document each of the constants. And that makes it a little bit more easier to understand from the outside what this package is supposed to do.